frontend/pages/policies/ManagePoliciesPage/components/InstallSoftwareModal/InstallSoftwareModal.tsx

```diff
@@ -206,13 +206,45 @@ const InstallSoftwareModal = ({
   const memoizedAvailableSoftwareOptions = useMemo(() => {
     const cache = new Map();
     return (policy: IFormPolicy) => {
-      const key = policy.platform;
+      let options = availableSoftwareOptions(policy) || [];
+      const installOptionsByPlatformMismatchSelectedInstaller =
+        policy.swIdToInstall &&
+        !options.some((opt) => opt.value === policy.swIdToInstall);
+
+      // More unique cache key if installOptionsByPlatformMismatchSelectedInstaller
+      const key = `${policy.platform}${
+        installOptionsByPlatformMismatchSelectedInstaller
+          ? `-${policy.swIdToInstall}`
+          : ""
+      }`;
       if (!cache.has(key)) {
-        cache.set(key, availableSoftwareOptions(policy));
+        // Add the current software if it's not in the options
+        // due to user-created a platform mismatch
+        if (installOptionsByPlatformMismatchSelectedInstaller) {
+          const currentSoftware = titlesAvailableForInstall?.find(
+            (title) => title.id === policy.swIdToInstall
+          );
+          if (currentSoftware) {
+            options = [
+              {
```

Comment:
This feels like it won't work with VPP?

Probably need to extract L176-200 to get this to work, as someone could set a VPP app and then swap the policy to query on Windows.

Reply:
done!
<img width="789" alt="Screenshot 2025-01-15 at 5 54 59 PM" src="https://github.com/user-attachments/assets/dafc9b84-a8a8-4c47-97eb-e3ffde3cb2d5" />